client/file-manager/src/rocksdb.rs

```diff
@@ -42,35 +42,14 @@ fn open_or_creating_rocksdb(db_path: String) -> io::Result<Database> {
 }
 
 /// Storage backend for RocksDB.
-pub struct StorageDb<Hasher> {
+pub struct StorageDb<T> {
     pub db: Arc<dyn KeyValueDB>,
```

Comment:
Can this `dyn` be removed somehow or is it overcomplicating too much?

Reply:
AFAIU we still need it to use the `In Memory implementation` of the `KeyValueDB` trait in the tests. Although it's known at compile time, I wasn't able to make it work with static dispatch instead of dynamic dispatch 🤔

Reply:
Actually, an update, I was able to remove the dynamic dispatch and use static dispatch instead. Will leave this improvement to another PR, though.